src/components/WalletButton.tsx

```diff
@@ -0,0 +1,58 @@
+
+import { Button, Layout, Modal } from '@stellar/design-system';
+import { useWallet } from '../hooks/useWallet';
+import { connectWallet, disconnectWallet } from '../util/wallet';
+import { useState } from 'react';
+
+
+export const WalletButton = () => {
+  const [showModal, setShowModal] = useState(false);
+  const { address, setAddress } = useWallet()
+  const buttonLabel = address ? `${address.slice(0, 10)}...` : "Connect"
+
+  // TODO: figure out why `useWallet` is getting mounted/used twice
+
+  const handleClick = () => {
+    void (async () => {
+      if (address) {
+        setShowModal(true);
+      } else {
+        await connectWallet();
+      }
+    })();
+  };
+
+
+  return <Layout.Content>
+    <div id="modalContainer">
+      <Modal visible={showModal} onClose={() => setShowModal(false)} parentId="modalContainer">
+        <Modal.Heading>Do you want to disconnect ?</Modal.Heading>
+        <Modal.Footer itemAlignment='stack'>
+          <Button
+            size="md"
+            variant="primary"
+            onClick={ () => {
+              void(async () => {
+                await disconnectWallet();
+                setShowModal(false);
+                if (setAddress) setAddress(undefined)
+              })()}
+            }
+          >
+            Disconnect
+          </Button>
+          <Button
+            size="md"
+            variant="tertiary"
+            onClick={() => {
+              setShowModal(false);
+            }}
+          >
+            Cancel
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    </div>
+    <Button variant="primary" size="lg" onClick={handleClick}>{buttonLabel}</Button>
```

Comment:
You might want to use https://design-system.stellar.org/profile here instead, it will handle displaying the address for you

Reply:
Great idea! Done.